### ISSUE TITLE
Multiple JSON Schemas - Selection of single Schema not working

### ISSUE THREAD

**Issue by BlockyTheDev** [redhat-developer/vscode-yaml#983]
## Describe the bug
It is not possible to select one JSON schema of the list, if multiple are detected for a specific file.

Detected Schemas:
https://github.com/SchemaStore/schemastore/blob/404ada34125022391ae5a4484d490f3d769a6555/src/schemas/json/bungee-plugin.json
https://github.com/SchemaStore/schemastore/blob/404ada34125022391ae5a4484d490f3d769a6555/src/schemas/json/bukkit-plugin.json
https://github.com/SchemaStore/schemastore/blob/404ada34125022391ae5a4484d490f3d769a6555/src/schemas/json/pocketmine-plugin.json

## Expected Behavior
The selection should work without problems.

## Current Behavior
This VSCode extension is detecting three schemas for the same file, what is is right due to the same file name... of the file. In the right lower corner it shows "Multiple JSON Schemas...", when I click on this, a menu opens with the title "Select JSON schema", but when I'm going to select one out of the list nothing is happening. Still the same selection of all three files.

## Steps to Reproduce
1. Create a file named `plugin.yml`
2. Click the button "Multiple JSON Schemas..."
3. Select one, for example "Bukkit plugin.yml"
4. Look at the button and see, that the button didn't change and the selection is still the same.

## Environment
- [X] Windows
- [ ] Mac
- [ ] Linux
- [ ] other (please specify)

**Comment by confiq** [comment]
Thank you for this ticket.

**Comment by ephb** [comment]
I have also encountered this. Where would the information about what the current schema is be stored?

**Comment by yaroslavsadin** [comment]
+1, can't select single JSON schema

**Comment by nikola418** [comment]
> Thank you for this ticket.
> 
> I've fixed by editing `.vscode/settings.json` file

Yeah, that's assuming that it works. I removed all the schemas from my settings and I can't get the extension to unselect the schemas rendering the whole extension useless.

**Comment by vishalnandagopal** [comment]
Bump, issue still exists

**Comment by jackytank** [comment]
+1 the issue still persist, detected by using continue.dev extension config.yaml. meanwhile I also check the vscode's settings.json to resolve the issue but the issue still persist. The version of this extension is 1.0.14 and I also check in C://Users/<myuser>/.vscode/extensions/ and those 2 version 1.1.49 & 1.0.13 didn't exist but VSCode Schema still appear 3 schemas

![Image](https://github.com/user-attachments/assets/584b70dc-9a2d-42f8-8e7e-04c1af5cabac)

![Image](https://github.com/user-attachments/assets/c31b4c3e-3f52-419a-adc9-011da3038aaa)

**Comment by jackytank** [comment]
I fixed my multiple 3 yaml-schema.json selected issue by stating this line at first line in my .yaml I wish to validating

![Image](https://github.com/user-attachments/assets/1bff87cc-9a81-43f5-b60b-7bf21f12a69b)

**Comment by kpocius** [comment]
@louis-vines, it does help! Thanks for the tip 🙏🏻 In my case the issue was precisely that I had a bunch of hardcoded definitions in workspace.json (no idea how).
